scripts/creation-portal-migration/template/generate/gQuestionSet.js

```diff
@@ -0,0 +1,1294 @@
+                reasons: "",
+              },
+            },
+          });
+          // Add the migrated question to the section's children array
+          sectionData.children = [
+            ...sectionData.children,
+            migratedQuestion?.referenceQuestionId,
+          ];
+          // Add the original question ID to the section's question IDs array
+          sectionData.questionIds = [...sectionData.questionIds, qid];
+          // Add the migrated question's metadata to the section's nodesModified property
+          sectionData.nodesModified[migratedQuestion?.referenceQuestionId] = {
+            isNew: false,
+            metadata: {
+              ...omit(migratedQuestion, "referenceQuestionId"),
+              visibility: "Parent",
+            },
+            objectType: "Question",
+            root: false,
+          };
+          migratedQuestion = omit(migratedQuestion, "referenceQuestionId");
+          // Update the sections object with the updated section data
+          sections[sectionData?.sectionId] = {
+            ...sections[sectionData?.sectionId],
+            ...sectionData,
+          };
+        }
+      }
+    }
+  }
+
+  return { sections, migratedCount };
+};
+
+/**
+ * To check exactly in which section the question present for nonmatrix
+ * @method
+ * @name getQueCriteriaIdAndData
+ * @param {String} qid - nonMatrixQuestionIds (objectId's) in mongo
+ * @param {Object[]} sections - newly mapped sections to update the hierarchy
+ * @returns {JSON} returns the question section
+ **/
+const getQueCriteriaIdAndData = (qid, sections) => {
+  const sectionIds = Object.keys(sections);
+  let criteriaData = {};
+  sectionIds.map((sectionId) => {
+    if (sections[sectionId]?.questionIds?.includes(qid)) {
+      criteriaData = sections[sectionId];
+    }
+  });
+  return criteriaData;
+};
+
+/**
+ * To check exactly in which section the question present for matrix
+ * @method
+ * @name getMatrixQueCriteriaIdAndData
+ * @param {String} qid - MatrixQuestionIds (objectId's) in mongo
+ * @param {Object[]} sections - newly mapped sections to update the hierarchy
+ * @returns {JSON} returns the question section
+ **/
+const getMatrixQueCriteriaIdAndData = (qid, sections) => {
+  const sectionIds = Object.keys(sections);
+  let criteriaData = {};
+  sectionIds.map((sectionId) => {
+    if (
+      sections[sectionId]?.type === "matrix" &&
+      sections[sectionId]?.instanceQuestions.includes(qid)
+    ) {
+      criteriaData = sections[sectionId];
+    }
+  });
+  return criteriaData;
+};
+
+/**
+ * Get the instanceQuestions and children questions if it's a matrix question
+ * @method
+ * @name getInstanceQuestionIds
+ * @param {Object} question - question
+ * @param {String} type - type
+ * @returns {JSON} returns all the instance and children questionIds belongs to matrix question
+ **/
+const getInstanceQuestionIds = (question, type = "") => {
+  let instanceQuestionsIds = [];
+  if (type === "instance") {
```

Comment:
@jayesh9747 please add all strings to common. Do this for entire string in migration folder